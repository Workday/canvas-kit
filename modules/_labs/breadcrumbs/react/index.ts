import Breadcrumbs from './lib/Breadcrumbs';

export default Breadcrumbs;
export {Breadcrumbs};
export * from './lib/Breadcrumbs';
