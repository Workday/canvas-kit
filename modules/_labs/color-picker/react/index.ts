export {default, default as ColorPicker, ColorPickerProps} from './lib/ColorPicker';
export {Swatch} from './lib/parts/Swatch';
