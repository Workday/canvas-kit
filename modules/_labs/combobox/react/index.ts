import Combobox from './lib/Combobox';

export default Combobox;
export {Combobox};
export * from './lib/Combobox';
