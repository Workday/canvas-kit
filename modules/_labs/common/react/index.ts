export * from './lib/Flex';
export * from './lib/Box';
