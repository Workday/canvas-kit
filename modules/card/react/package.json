{
  "name": "@workday/canvas-kit-react-card",
  "version": "3.4.0",
  "description": "Common Canvas card component for React",
  "author": "Workday, Inc. (https://www.workday.com)",
  "license": "Apache-2.0",
  "main": "dist/commonjs/index.js",
  "module": "dist/es6/index.js",
  "sideEffects": false,
  "types": "dist/es6/index.d.ts",
  "repository": {
    "type": "git",
    "url": "http://github.com/Workday/canvas-kit/tree/master/modules/card/react"
  },
  "files": [
    "dist/",
    "lib/",
    "index.ts"
  ],
  "scripts": {
    "watch": "yarn build:es6 -w",
    "test": "echo \"Error: no test specified\" && exit 1",
    "clean": "rimraf dist && rimraf .build-info && mkdirp dist",
    "build:cjs": "tsc -p tsconfig.cjs.json",
    "build:es6": "tsc -p tsconfig.es6.json",
    "build:rebuild": "npm-run-all clean build",
    "build": "npm-run-all --parallel build:cjs build:es6",
    "depcheck": "node ../../../utils/check-dependencies-exist.js"
  },
  "keywords": [
    "canvas",
    "canvas-kit",
    "react",
    "components",
    "workday",
    "card"
  ],
  "peerDependencies": {
    "react": ">= 16.8 < 17"
  },
  "dependencies": {
    "@emotion/core": "^10.0.21",
    "@emotion/styled": "^10.0.17",
    "@workday/canvas-kit-react-core": "^3.4.0"
  }
}
