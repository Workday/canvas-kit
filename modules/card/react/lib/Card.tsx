import * as React from 'react';
import styled from '@emotion/styled';
import isPropValid from '@emotion/is-prop-valid';
import {
  colors,
  depth as depthValues,
  type,
  spacing,
  borderRadius,
  CanvasDepthValue,
  CanvasSpacingValue,
} from '@workday/canvas-kit-react-core';

export interface CardProps extends React.HTMLAttributes<HTMLDivElement> {
  /**
   * The heading of the Card.
   */
  heading?: React.ReactNode;

  /**
   * The id of the Card heading. Tie this to an `aria-labelledby` for accessibility.
   */
  headingId?: string;

  /**
   * The padding of the Card. Imported from `@workday/canvas-kit-react-core`.
   * @default spacing.l
   */
  padding?: CanvasSpacingValue;

  /**
   * The depth of the Card. Imported from `@workday/canvas-kit-react-core`.
   * @default depthValues[2]
   */
  depth?: CanvasDepthValue;

  /**
   * The width of the Card.
   */
  width?: number | string;

  /**
   * The height of the Card.
   */
  height?: number | string;
}

const Box = styled('div', {
  shouldForwardProp: prop => isPropValid(prop) && prop !== 'width' && prop !== 'height',
})<CardProps>(
  {
    backgroundColor: colors.frenchVanilla100,
    border: `1px solid ${colors.soap500}`,
    borderRadius: borderRadius.l,
    boxSizing: 'border-box',
  },
  ({depth}) => depth,
  ({padding}) => ({padding}),
  ({width}) => width && {width},
  ({height}) => height && {height}
);

const Header = styled('h3')(type.h3, {
  marginBottom: spacing.m,
  marginTop: 0,
});

const Body = styled('div')(type.body);

export default class Card extends React.Component<CardProps> {
  public render() {
    const {
      depth = depthValues[2],
      padding = spacing.l,
      heading,
      headingId,
      ...elemProps
    } = this.props;

    return (
      <Box {...elemProps} depth={depth} padding={padding}>
        {heading && <Header id={headingId}>{heading}</Header>}
        <Body>{this.props.children}</Body>
      </Box>
    );
  }
}
