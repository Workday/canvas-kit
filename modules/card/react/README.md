# Canvas Kit Card

Generic Canvas card component

[> Workday Design Reference](https://design.workday.com/components/containers/cards)

## Installation

```sh
yarn add @workday/canvas-kit-react
```

or

```sh
yarn add @workday/canvas-kit-react-card
```

## Usage

```tsx
import Card from '@workday/canvas-kit-react-card';
import {depth, spacing} from '@workday/canvas-kit-react-core';

<Card heading="Title">
  Card contents
</Card>

<Card padding={spacing.l}>
  Card with custom padding
</Card>

<Card depth={depth[1]}>
  Card with custom depth
</Card>
```

## Static Properties

> None

## Component Props

### Required

> None

### Optional

#### `heading: ReactNode`

> Heading at the top of the card.

Default: `null`

---

#### `headingId: string`

> Heading ID for accessibility. Useful tie to an `aria-labelledby`

---

#### `padding: 0 | CanvasSpacingValue`

> Padding of the card. Style imported from `@workday/canvas-kit-react-core`.

Default: `spacing.l`

---

#### `depth: CanvasDepthValue`

> Depth of the card. Style imported from `@workday/canvas-kit-react-core`.

Default: `depth[2]`

---

#### `width: number | string`

> Width of the card.

Default: `null`

---

#### `height: number | string`

> Height of the card.

Default: `null`
