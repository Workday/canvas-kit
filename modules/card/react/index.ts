import Card from './lib/Card';

export default Card;
export {Card};
export * from './lib/Card';
