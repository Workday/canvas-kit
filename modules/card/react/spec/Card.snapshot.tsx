import * as React from 'react';
import Card from '../lib/Card';
import * as renderer from 'react-test-renderer';

describe('Card Snapshots', () => {
  test('renders as expected', () => {
    const component = renderer.create(<Card>Card</Card>);
    expect(component).toMatchSnapshot();
  });

  test('renders a card with depth', () => {
    const component = renderer.create(
      <Card
        depth={{
          boxShadow: '0px 2px 4px 0 rgba(0, 0, 0, 0.08)' as '0px 2px 4px 0 rgba(0, 0, 0, 0.08)',
        }}
      >
        Card with custom depth
      </Card>
    );
    expect(component).toMatchSnapshot();
  });

  test('renders a card with heading', () => {
    const component = renderer.create(<Card heading="Card Title">Card with a heading</Card>);
    expect(component).toMatchSnapshot();
  });

  test('renders a card with padding', () => {
    const component = renderer.create(<Card padding="12px">Card with custom padding</Card>);
    expect(component).toMatchSnapshot();
  });

  test('renders a card with width', () => {
    const component = renderer.create(<Card width="12px">Card with custom width</Card>);
    expect(component).toMatchSnapshot();
  });

  test('renders a card with height', () => {
    const component = renderer.create(<Card height="12px">Card with custom width</Card>);
    expect(component).toMatchSnapshot();
  });
});
