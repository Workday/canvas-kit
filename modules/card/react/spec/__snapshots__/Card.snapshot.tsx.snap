// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Snapshots renders a card with depth 1`] = `
.emotion-0 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-2 {
  background-color: #ffffff;
  border: 1px solid #ced3d9;
  border-radius: 8px;
  box-sizing: border-box;
  box-shadow: 0px 2px 4px 0 rgba(0,0,0,0.08);
  padding: 32px;
}

<div
  className="emotion-2 emotion-3"
>
  <div
    className="emotion-0 emotion-1"
  >
    Card with custom depth
  </div>
</div>
`;

exports[`Card Snapshots renders a card with heading 1`] = `
.emotion-4 {
  background-color: #ffffff;
  border: 1px solid #ced3d9;
  border-radius: 8px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 32px;
}

.emotion-2 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-0 {
  font-size: 20px;
  line-height: 28px;
  font-weight: 700;
  color: #333333;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  margin-bottom: 24px;
}

<div
  className="emotion-4 emotion-5"
>
  <h3
    className="emotion-0 emotion-1"
  >
    Card Title
  </h3>
  <div
    className="emotion-2 emotion-3"
  >
    Card with a heading
  </div>
</div>
`;

exports[`Card Snapshots renders a card with height 1`] = `
.emotion-0 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-2 {
  background-color: #ffffff;
  border: 1px solid #ced3d9;
  border-radius: 8px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 32px;
  height: 12px;
}

<div
  className="emotion-2 emotion-3"
  height="12px"
>
  <div
    className="emotion-0 emotion-1"
  >
    Card with custom width
  </div>
</div>
`;

exports[`Card Snapshots renders a card with padding 1`] = `
.emotion-0 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-2 {
  background-color: #ffffff;
  border: 1px solid #ced3d9;
  border-radius: 8px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 12px;
}

<div
  className="emotion-2 emotion-3"
>
  <div
    className="emotion-0 emotion-1"
  >
    Card with custom padding
  </div>
</div>
`;

exports[`Card Snapshots renders a card with width 1`] = `
.emotion-0 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-2 {
  background-color: #ffffff;
  border: 1px solid #ced3d9;
  border-radius: 8px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 32px;
  width: 12px;
}

<div
  className="emotion-2 emotion-3"
  width="12px"
>
  <div
    className="emotion-0 emotion-1"
  >
    Card with custom width
  </div>
</div>
`;

exports[`Card Snapshots renders as expected 1`] = `
.emotion-2 {
  background-color: #ffffff;
  border: 1px solid #ced3d9;
  border-radius: 8px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 32px;
}

.emotion-0 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

<div
  className="emotion-2 emotion-3"
>
  <div
    className="emotion-0 emotion-1"
  >
    Card
  </div>
</div>
`;
